qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/FieldCapsIT.java

```diff
@@ -252,4 +262,46 @@ public void testAllIndicesWithIndexFilter() throws Exception {
         assertThat(resp.getField("blue_field").keySet(), contains("keyword"));
         assertTrue(resp.getField("blue_field").get("keyword").isSearchable());
     }
+
+    @SuppressWarnings("unchecked")
+    private RestClient getUpgradedNodeClient() throws IOException {
+        for (HttpHost host : getClusterHosts()) {
+            RestClient client = RestClient.builder(host).build();
+            Request nodesRequest = new Request("GET", "_nodes/_local/_none");
+            Map<String, ?> nodeMap = (Map<String, ?>) entityAsMap(client.performRequest(nodesRequest)).get("nodes");
+            Map<String, ?> nameMap = (Map<String, ?>) nodeMap.values().iterator().next();
+            String version = (String) nameMap.get("version");
+            if (version.equals(Version.CURRENT.toString())) {
+                return client;
+            }
+            client.close();
+        }
+        throw new IllegalStateException("Couldn't find node on version " + Version.CURRENT);
+    }
+
+    // Test field type filtering on mixed cluster
+    public void testAllIndicesWithFieldTypeFilter() throws Exception {
+        assumeFalse("required mixed or upgraded cluster", CLUSTER_TYPE == ClusterType.OLD);
+        RestClient restClient = getUpgradedNodeClient();
```

Comment:
add a comment to these two methods to explain why you need a special client and what the expectations are? e.g. only newer nodes support the new parameter, but we emulate the filtering on the coordinating node hence it does not matter that other nodes are on previous versions